README.md

```diff
@@ -242,7 +242,80 @@ Once the tests are DeFined, running `npm run build-snapshots -- -p <protocol-nam
+### Write Actions for adapters
+
+Product adapters can implement the `getTransactionParams` method to return an object that can be used to form a transaction (usually `to` and `data`).
+
+#### Determine the actions supported by the product adapter
+First of all, it is necessary to determine what actions that product adapter supports. There's a list of available actions which can be extended when the needs arise (entries need to be added to [this constant declaration](src/types/writeActions.ts). The current supported actions are:
+- Deposit
+- Withdraw
+- Borrow
+- Repay
+
+#### Create input schemas for every action
+After that, an object called `WriteActionInputs` needs to be exported from that adapter file. The object includes a list of the available actions and the respective Zod schemas for the input.
+
+We use Zod schemas because it is a popular library that allows us to create appropriate types in the background to validate the input. Here is a [list of primitives that Zod supports](https://zod.dev/?id=primitives).
+
+That object needs to satisfy the type `WriteActionInputSchemas`. Here's a simple example implementation for an adapter that supported `deposit` and `withdraw`:
+
+```
+export const WriteActionInputs = {
+  [WriteActions.Deposit]: z.object({
+    asset: z.string(),
+    amount: z.string(),
+    onBehalfOf: z.string(),
+    referralCode: z.number(),
+  }),
+  [WriteActions.Withdraw]: z.object({
+    asset: z.string(),
+    amount: z.string(),
+    to: z.string(),
+  }),
+} satisfies WriteActionInputSchemas
+```
+
+After creating that input schema, and making sure that the project is building with `npm run build:watch`, it is necessary to run `npm run build-types` so that the types and schemas are correctly generated.
+
+#### Implement getTransactionParams method logic
+Implement the `getTransactionParams` method within your adapter.
+
+The easier way is to set a switch statement using the `WriteAction` as discriminator, this should allow TypeScript language server to determine the correct type within the switch statement entry.
+
+```
+async getTransactionParams({
+  action,
+  inputs,
+}: Extract<
+  GetTransactionParams,
+  { protocolId: typeof Protocol.YourProtocolKey; productId: 'YourProductId' }
+>): Promise<{ to: string; data: string }> {
+  switch (action) {
+    case WriteActions.Deposit: {
+      const { asset, amount, onBehalfOf, referralCode } = inputs
+      return poolContract.supply.populateTransaction(
+        asset,
+        amount,
+        onBehalfOf,
+        referralCode,
+      )
+    }
+    case WriteActions.Withdraw: {
+      const { asset, amount, to } = inputs
+      return poolContract.withdraw.populateTransaction(asset, amount, to)
+    }
+  }
+}
+```
+
+#### Video tutorial
+Check out the tutorial video below for an intro to on how to build write adapter actions:
+
+[![DeFi Adapter Write Actions Tutorial](video-thumbnail.png)](https://drive.google.com/file/d/1ZNWwOkzHlc7Zt2qLy5ZRuHqoDgWSnww7/view)
+
+
```

Comment:
can you add this to the 🎥 DeFi Adapter Tutorial section please.

So the end result will be 

🎥 DeFi Adapter Tutorial


Step 1 Create read adapter 
[video ]
Step 2 Add write adapter support to read adapter
[video]